packages/input/src/presenters/InputPresenter.js

```diff
@@ -36,18 +41,33 @@ export default function InputPresenter(props) {
               metadata.colorSchemeId
             )
           : styles;
+
+        let className = css(cssStyles.input);
+        const { className: customClassName } = otherProps;
+        if (customClassName) {
+          const elementClassName = customClassName
+            .split(" ")
+            .reduce((acc, cur) => cx(acc, `${cur.trim()}__input`), "");
+
+          className = cx(
+            className,
+            { [customClassName]: variant === variants.PLAIN },
```

Comment:
Sorry for another change, I didn't realize that the `plain` variant does not have the halo either and is just the `input`, so since we are checking for that `variant` maybe we should just use that for the logic that if it is `plain` then we only pass in what the user passes in and not `__input`.

Reply:
Sure, i've updated on the changes. Just one concern that as the `split` is only a function for `String`, if user passes other type values rather than a string, errors will be thrown. Or we can just leave this to the users, as they need pass correct className as string?

Reply:
I think we can leave it up to the user to pass in a string for the `className`.